Source/WebCore/css/CSSSelector.cpp

```diff
@@ -861,6 +874,9 @@ String CSSSelector::selectorText(StringView separator, StringView rightSide) con
             break;
         }
         return previousSelector->selectorText(separator, builder);
+    } else if (!separatorTextForNestingRelative().isNull()) {
+        // We have a separator but no tag history which can happen with implicit relative nesting selector
+        return separatorTextForNestingRelative() + builder.toString();
```

Comment:
it would be more stylish to not call separatorTextForNestingRelative() twice, for example.
```
} else if (auto separatorText = separatorTextForNestingRelative(); !separatorText.isNull()) 
    return separatorText + builder.toString();
```